packages/atomic/src/components/search/atomic-result-date/atomic-result-date.spec.ts

```diff
@@ -0,0 +1,239 @@
+import './atomic-result-date';
+
+vi.mock('@coveo/headless', {spy: true});
+
+describe('atomic-result-date', () => {
+  let i18n: i18n;
+  let mockResult: Result;
+
+  beforeEach(async () => {
+    console.error = vi.fn();
+
+    i18n = await createTestI18n();
+    i18n.addResourceBundle(
+      'en',
+      'translation',
+      {
+        'calendar-same-day': '[Today]',
+        'calendar-next-day': '[Tomorrow]',
+        'calendar-next-week': 'dddd',
+        'calendar-last-day': '[Yesterday]',
+        'calendar-last-week': '[Last] dddd',
+      },
+      true
+    );
+
+    mockResult = buildFakeResult({
+      title: 'Test Result',
+      raw: {
+        date: '2021/09/03@10:31:23',
+        invalidDate: 'not-a-date',
+        urihash: '',
+      },
+    });
+  });
+
+  const renderComponent = async (
+    options: {
+      field?: string;
+      format?: string;
+      relativeTime?: boolean;
+      result?: Result | null;
+    } = {}
+  ) => {
+    const resultToUse = 'result' in options ? options.result : mockResult;
+    const {element, atomicInterface} =
+      await renderInAtomicResult<AtomicResultDate>({
+        template: html`<atomic-result-date
+          field=${ifDefined(options.field)}
+          format=${ifDefined(options.format)}
+          ?relative-time=${options.relativeTime}
+        ></atomic-result-date>`,
+        selector: 'atomic-result-date',
+        result: resultToUse === null ? undefined : resultToUse,
+        bindings: (bindings) => {
+          bindings.i18n = i18n;
+          bindings.store = {
+            ...bindings.store,
+            onChange: vi.fn(),
+            state: {
+              ...bindings.store?.state,
+              loadingFlags: [],
+            },
+          };
+          // Mock interfaceElement with language property
+          bindings.interfaceElement = {
+            ...bindings.interfaceElement,
+            language: 'en',
+            // biome-ignore lint/suspicious/noExplicitAny: mock interface element
+          } as any;
+          return bindings;
+        },
+      });
+
+    await atomicInterface.updateComplete;
+    await element?.updateComplete;
+
+    return element;
+  };
+
+  it('should be defined', () => {
+    const el = document.createElement('atomic-result-date');
+    expect(el).toBeInstanceOf(AtomicResultDate);
+  });
+
+  describe('when the field is valid', () => {
+    it('should render the date with default format', async () => {
+      const element = await renderComponent({field: 'date'});
+
+      expect(element).toBeDefined();
+      expect(element.textContent?.trim()).toBe('3/9/2021');
+    });
+
+    it('should render the date with custom format', async () => {
+      const element = await renderComponent({
+        field: 'date',
+        format: 'YYYY-MM-DD',
+      });
+
+      expect(element).toBeDefined();
+      expect(element.textContent?.trim()).toBe('2021-09-03');
+    });
+  });
+
+  describe('when the field does not exist', () => {
```

Comment:
combine `it` and `describe`.

similar comment in this spec file when you have conditions with a single test.